packages/app-builder-lib/src/util/NodeModuleCopyHelper.ts

```diff
@@ -130,17 +133,29 @@ export class NodeModuleCopyHelper extends FileCopyHelper {
         CONCURRENCY
       )
 
+      let isEmpty = true
       for (const child of sortedFilePaths) {
         if (child != null) {
           result.push(child)
+          this.metadata.get(child)?.isSymbolicLink() && symlinkFiles.set(child, result.length - 1)
+          isEmpty = false
         }
       }
+      isEmpty && emptyDirs.add(dirPath)
 
       dirs.sort()
       for (const child of dirs) {
         queue.push(dirPath + path.sep + child)
       }
     }
-    return result
+
+    for (const [file, index] of symlinkFiles) {
+      const resolvedPath = realpathSync(file)
+      if (emptyDirs.has(resolvedPath)) {
+        // delete symlink file if target is a empty dir
+        result[index] = ""
```

Comment:
Minor nit: Can we use `undefined` here to avoid using an empty string as a false-y value?

Reply:
There can't be any empty strings in the result, as these are paths, and the paths passed in are not empty strings.

However, from a coding perspective, using `undefined` would be better.

Fixed it.